sdk/src/transaction/error.rs

```diff
@@ -157,6 +157,10 @@ pub enum TransactionError {
     /// LoadedAccountsDataSizeLimit set for transaction must be greater than 0.
     #[error("LoadedAccountsDataSizeLimit set for transaction must be greater than 0.")]
     InvalidLoadedAccountsDataSizeLimit,
+
+    /// Stake program is unavailable during reward period
+    #[error("Stake program is unavailable during reward period")]
+    StakeProgramUnavailable,
```

Comment:
Since this is a global `TransactionError`, my bike shed would be to drop the "Stake" entirely:
* `ProgramUnavailable({ account_index: u8})`
* `ProgramTemporarilyUnavailable({account_index: u8})`

Maybe there'll be times that programs will need to be restricted in the future as well, and this resolves the aesthetic issue with calling out a specific program in `TransactionError`

Reply:
This sounds like a good suggestion to me.

Reply:
sounds good to me too.

Reply:
`ProgramExecutionTemporarilyRestricted{account_index: u8}` and i'm game.

motivation being to disambiguate _why_ the program is unavailable

runtime could return this during the one slot deployment cool off period too

Reply:
I think I like `Temporarily`. I think we need only 1 of Michael's 2 suggestions, right?

Reply:
>  I think we need only 1 of Michael's 2 suggestions, right?

Yep. Sounds like we all agree on `ProgramExecutionTemporarilyRestricted`

Reply:
rebased to fix merge with other errors, then added a commit for this rename.